src/components/Modal/index.tsx

```diff
@@ -0,0 +1,73 @@
+import * as Dialog from '@radix-ui/react-dialog'
+import { transparentize } from 'polished'
+import { ReactNode } from 'react'
+import styled, { css } from 'styled-components'
+import { ReactComponent as CloseIcon } from '@assets/close.svg'
+
+export type ModalProps = {
+  trigger: ReactNode
+  children: ReactNode
+}
+
+const Overlay = styled(Dialog.Overlay)`
+  ${({ theme }) => css`
+    position: fixed;
+    inset: 0;
+    background-color: ${transparentize(0.6, theme.colors.black[100])};
+  `}
+`
+
+const Content = styled(Dialog.Content)`
+  ${({ theme }) => css`
+    border-radius: 4px;
+    max-width: 40rem;
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    padding: 16px;
+    position: fixed;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    background-color: ${theme.colors.white[100]};
+  `}
+`
+
+const CloseWrapper = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`
+
+const IconButton = styled.button`
+  ${({ theme }) => css`
+    all: unset;
+    padding: 6px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: ${transparentize(0.8, theme.colors.grey[300])};
+    border-radius: 50%;
+    cursor: pointer;
+  `}
+`
+
+export const Modal = ({ trigger, children }: ModalProps) => {
+  return (
+    <Dialog.Root>
```

Comment:
O componente do radix, contem subcomponents Title e Description, seria legal reexportar eles pra usar quando for compor um modal como é o caso do modal de sign in ou sign up, saca? Ele torna as coisas mais acessíveis.